File: regression-harness/selftest-corpus/dtofields-only/.idl-regression/sample_app.ts
// Hand-rolled minimal sample app for M4 acceptance of idl-regress (TS adapter).
// Covers the same 3 fixtures as the Scala sample app (sample_app.scala):
//   idltest.dtofields.IntPair.Struct × 2 (zero + mixed)
//   idltest.dtofields.WHPair.Struct  × 1 (default)
// A future LLM-generated sample app supersedes this.

import { IntPairStruct } from './idltest/dtofields/IntPair';
import { WHPairStruct }  from './idltest/dtofields/WHPair';

function emit(wireId: string, scenario: string, value: { serialize(): unknown }): void {
  console.log(`${wireId}\t${scenario}\t${JSON.stringify(value.serialize())}`);
}

// idltest.dtofields.IntPair.Struct — two scenarios.
emit('idltest.dtofields.IntPair.Struct', 'zero',  new IntPairStruct({ x: 0,  y: 0  }));
emit('idltest.dtofields.IntPair.Struct', 'mixed', new IntPairStruct({ x: 17, y: -3 }));

// idltest.dtofields.WHPair.Struct — one scenario.
emit('idltest.dtofields.WHPair.Struct',  'default', new WHPairStruct({ w: 640, h: 480 }));

File: regression-harness/selftest-corpus/dtofields-only/.idl-regression/sample_app.cs
// Hand-rolled minimal sample app for M5 acceptance of idl-regress (C# adapter).
// Covers the same 3 fixtures as the Scala/TS sample apps:
//   idltest.dtofields.IntPair.Struct × 2 (zero + mixed)
//   idltest.dtofields.WHPair.Struct  × 1 (default)
// A future LLM-generated sample app supersedes this.

using System;
using Newtonsoft.Json;
using Idltest.Dtofields;

namespace IdlRegress.CSharpDriver {
  public static class Program {
    private static readonly JsonSerializerSettings S = new JsonSerializerSettings {
      NullValueHandling = NullValueHandling.Ignore,
      Formatting = Formatting.None,
    };

    public static int Main(string[] args) {
      // idltest.dtofields.IntPair.Struct — two scenarios.
      Emit("idltest.dtofields.IntPair.Struct", "zero",    new IntPairStruct(0,  0));
      Emit("idltest.dtofields.IntPair.Struct", "mixed",   new IntPairStruct(17, -3));
      // idltest.dtofields.WHPair.Struct — one scenario.
      Emit("idltest.dtofields.WHPair.Struct",  "default", new WHPairStruct(640, 480));
      return 0;
    }

    private static void Emit(string wireId, string scenario, object v) {
      var json = JsonConvert.SerializeObject(v, S);
      Console.WriteLine($"{wireId}\t{scenario}\t{json}");
    }
  }
}
